### ISSUE TITLE
NVDA Fails to Detect Layout Switcher Losing Focus in Certain Scenarios

### ISSUE THREAD

**Issue by SavagingSky** [nvaccess/nvda#17872]
### Steps to reproduce:
1. Focus desktop or any window. Note: I've tried to reproduce this behavior while an explorer window is open (some folder for example), but the results are unreliable. Sometimes errors are reproduced, sometimes not. Also, nvda behaves as it should and reenteres the previously focused window.
2. Switch layout (in my case it's alt+shift), release the keys (important!).
3. Hold the first key (in my case it's left alt).
4. Press the second key (in my case it's right shift).
5. Release the second key (right shift in my case).
6. Release the first key.
### Actual behavior:

Nvda is in a state where it thinks (presumably) that the layout switcher flyout is still focused, which causes the following (assuming desktop was focused earlier):
1. Pressing any arrow key which sets a focus to any desktop item causes nvda to say desktop list, announcing the item afterwards.
2. Pressing any key except arrows is treated strangely by nvda, for example pressing letters on the desktop produces an error. If there is only one item starting with the pressed letter, nvda is silent and does not regain desktop focus.
If in browse mode (in my case in google chrome), pressing the escape key in such a state causes nvda to pass through this key to the application directly, instead of just exiting the browse mode.
### Expected behavior:

Nvda should not behave like this. Instead, layout switcher losing focus should be treated correctly and non of the described in the expected behavior section should happen.
### NVDA logs, crash dumps and other attachments:
Reproducing the issue gives errors after nvda failed to register layout switcher losing focus, and I found two variations of those errors.
<details>
<summary>first variation</summary>
ERROR - eventHandler.executeEvent (15:49:14.663) - MainThread (6860):
error executing event: UIA_elementSelected on <NVDAObjects.Dynamic_VirtualContextMenuListItemUIA object at 0x0ABC46F0> with extra args of {}
Traceback (most recent call last):
File "eventHandler.pyc", line 353, in executeEvent
File "eventHandler.pyc", line 119, in __init__
File "eventHandler.pyc", line 128, in next
File "appModules\explorer.pyc", line 562, in event_UIA_elementSelected
File "eventHandler.pyc", line 128, in next
File "NVDAObjects\UIA\__init__.pyc", line 2401, in event_UIA_elementSelected
File "NVDAObjects\__init__.pyc", line 1315, in event_selection
File "baseObject.pyc", line 59, in __get__
File "baseObject.pyc", line 167, in _getPropertyViaCache
File "NVDAObjects\UIA\__init__.pyc", line 2372, in _get_controllerFor
File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (15:49:14.676) - MainThread (6860):
error executing event: UIA_elementSelected on <NVDAObjects.Dynamic_VirtualContextMenuListItemUIA object at 0x058F4A70> with extra args of {}
Traceback (most recent call last):
File "eventHandler.pyc", line 353, in executeEvent
File "eventHandler.pyc", line 119, in __init__
File "eventHandler.pyc", line 128, in next
File "appModules\explorer.pyc", line 562, in event_UIA_elementSelected
File "eventHandler.pyc", line 128, in next
File "NVDAObjects\UIA\__init__.pyc", line 2401, in event_UIA_elementSelected
File "NVDAObjects\__init__.pyc", line 1315, in event_selection
File "baseObject.pyc", line 59, in __get__
File "baseObject.pyc", line 167, in _getPropertyViaCache
File "NVDAObjects\UIA\__init__.pyc", line 2372, in _get_controllerFor
File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))
</details>
<details>
<summary>Second variation</summary>
ERROR - eventHandler.executeEvent (16:22:39.215) - MainThread (2668):
error executing event: focusEntered on <NVDAObjects.UIA.UIA object at 0x05B43670> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 353, in executeEvent
  File "eventHandler.pyc", line 119, in __init__
  File "eventHandler.pyc", line 128, in next
  File "NVDAObjects\__init__.pyc", line 1358, in event_focusEntered
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 1164, in _get_isPresentableFocusAncestor
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 2027, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 905, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1919, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (16:22:39.220) - MainThread (2668):
error executing event: focusEntered on <NVDAObjects.UIA.UIA object at 0x05B437F0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 353, in executeEvent
  File "eventHandler.pyc", line 119, in __init__
  File "eventHandler.pyc", line 128, in next
  File "NVDAObjects\__init__.pyc", line 1358, in event_focusEntered
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 1164, in _get_isPresentableFocusAncestor
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 2027, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 905, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1919, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (16:22:39.224) - MainThread (2668):
error executing event: focusEntered on <NVDAObjects.UIA.UIA object at 0x05003AF0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 353, in executeEvent
  File "eventHandler.pyc", line 119, in __init__
  File "eventHandler.pyc", line 128, in next
  File "NVDAObjects\__init__.pyc", line 1358, in event_focusEntered
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 1164, in _get_isPresentableFocusAncestor
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 2027, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 905, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1919, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (16:22:39.228) - MainThread (2668):
error executing event: focusEntered on <NVDAObjects.UIA.UIA object at 0x0A2060B0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 353, in executeEvent
  File "eventHandler.pyc", line 119, in __init__
  File "eventHandler.pyc", line 128, in next
  File "NVDAObjects\__init__.pyc", line 1358, in event_focusEntered
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 1164, in _get_isPresentableFocusAncestor
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 2027, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 905, in _get_presentationType
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1919, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (16:22:39.233) - MainThread (2668):
error executing event: gainFocus on <NVDAObjects.UIA.ListItem object at 0x0A206F90> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 353, in executeEvent
  File "eventHandler.pyc", line 119, in __init__
  File "eventHandler.pyc", line 128, in next
  File "appModules\explorer.pyc", line 496, in event_gainFocus
  File "eventHandler.pyc", line 128, in next
  File "NVDAObjects\UIA\__init__.pyc", line 1577, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1365, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1201, in reportFocus
  File "speech\speech.pyc", line 791, in speakObject
  File "speech\speech.pyc", line 831, in getObjectSpeech
  File "speech\speech.pyc", line 682, in getObjectPropertiesSpeech
  File "baseObject.pyc", line 59, in __get__
  File "baseObject.pyc", line 167, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1919, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))
</details>
I haven't found a way to reliably reproduce the second variation, but I think if the focus is on the desktop start button, for example, executing the actions in steps to reproduce does the trick.
Pressing any key (escape, letter) in this state produces the following:

ERROR - eventHandler.executeEvent (15:55:38.299) - MainThread (5324):
error executing event: typedCharacter on <NVDAObjects.Dynamic_VirtualContextMenuListItemUIA object at 0x05891BB0> with extra args of {'ch': '\x1b'}
Traceback (most recent call last):
File "eventHandler.pyc", line 353, in executeEvent
File "eventHandler.pyc", line 119, in __init__
File "eventHandler.pyc", line 128, in next
File "NVDAObjects\__init__.pyc", line 1253, in event_typedCharacter
File "speech\speech.pyc", line 1350, in speakTypedCharacters
File "api.pyc", line 362, in isTypingProtected
File "baseObject.pyc", line 59, in __get__
File "baseObject.pyc", line 167, in _getPropertyViaCache
File "NVDAObjects\__init__.pyc", line 1123, in _get_isProtected
File "baseObject.pyc", line 59, in __get__
File "baseObject.pyc", line 167, in _getPropertyViaCache
File "NVDAObjects\UIA\__init__.pyc", line 1919, in _get_states
File "NVDAObjects\UIA\__init__.pyc", line 1156, in _getUIACacheablePropertyValue
_ctypes.COMError: (-2147418094, (None, None, None, 0, None))

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2024.4.2
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.5608
#### Name and version of other software in use when reproducing the issue:
Google chrome v 134.0.6998.178 64 bit
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
I didn't try any other versions, but I remember that this behavior is present in earlier versions as well.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by SaschaCowley** [comment]
@josephsl any ideas here?

**Comment by Adriani90** [comment]
Duplicate of #9816 which is also reproducible on Windows 11.
See https://github.com/nvaccess/nvda/issues/9816#issuecomment-2501984916 for more details.
